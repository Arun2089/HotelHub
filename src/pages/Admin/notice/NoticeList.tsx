import { ChangeEvent, Fragment, MouseEvent, useState } from "react";
import {
  Paper,
  Typography,
  Divider,
  List,
  Box,
  Button,
  Skeleton,
  DialogContent,
} from "@mui/material";
import { Add as AddIcon } from "@mui/icons-material";

import NoticeListItem from "./NoticeListItem";
import NoticeModal from "./NoticeModal";
import AlertComponent from "@components/Alert";
import ConfirmationModal from "@components/ConfirmationModal";

import useDialog from "@src/hooks/useDialog.ts";
import useAlert from "@src/hooks/useAlert.ts";
import { sendData, deleteData, catchErrorMessage } from "@utils/index";
import { NoticeDataType, NoticeStateProps, SeverityType } from "@ts/types";
import { NOTICES_URL, METHOD } from "@constant/index";

const saveNotice = async (
  notice: NoticeDataType,
  handleAlert: (
    isOpen: boolean,
    message: string,
    serverity: SeverityType
  ) => void
) => {
  try {
    const { PUT, POST } = METHOD;
    const method = notice.id ? PUT : POST;
    const url = notice.id ? `${NOTICES_URL}/${notice.id}` : NOTICES_URL;
    const isDataSent = await sendData(url, method, notice);
    const message = isDataSent ? "Notice updated" : "Notice added";
    const severity = isDataSent ? "success" : "error";
    handleAlert(true, message, severity);
  } catch (error) {
    handleAlert(true, catchErrorMessage(error), "error");
  }
};

const deleteNotice = async (
  id: number | null,
  handleAlert: (
    isOpen: boolean,
    message: string,
    serverity: SeverityType
  ) => void
) => {
  try {
    if (!id) {
      throw new Error("notice do not exist");
    }
    const url = `${NOTICES_URL}/${id}`;
    const isDataDeleted = await deleteData(url);
    const message = isDataDeleted
      ? "Notice deleted"
      : "Error, Notice not deleted";
    const severity = isDataDeleted ? "success" : "error";
    handleAlert(true, message, severity);
  } catch (error) {
    handleAlert(true, catchErrorMessage(error), "error");
  }
};

type NoticesProps = {
  notices: NoticeDataType[] | null;
  setupdateNoticeCheck: React.Dispatch<React.SetStateAction<boolean>>;
};

const intitalNoticeState: NoticeDataType = {
  title: "",
  date: new Date().toLocaleDateString(),
  content: "",
};

const NoticeList = ({ notices, setupdateNoticeCheck }: NoticesProps) => {
  const [selectedNotice, setSelectedNotice] = useState<NoticeStateProps>({
    notice: intitalNoticeState,
    isModalOpen: false,
    isEditable: false,
    add: false,
  });
  const [deleteNoticeId, setDeleteNoticeId] = useState<number | null>(null);
  const { alert, handleAlert } = useAlert();
  const { isOpen, message, severity } = alert;

  const handleDialog = () => {
    deleteNotice(deleteNoticeId, handleAlert);
    setupdateNoticeCheck((val) => {
      return !val;
    });
  };
  const { open, handleDialogClick, handleDialogSubmit } =
    useDialog(handleDialog);

  const handleOpenNotice = (
    event: MouseEvent<Element>,
    state: string,
    noticeData: NoticeDataType
  ) => {
    event.stopPropagation();
    const isEditable = state === "edit";
    const add = state === "add";
    setSelectedNotice({
      notice: noticeData,
      isEditable,
      add,
      isModalOpen: true,
    });
  };

  const handleClose = (): void => {
    setSelectedNotice((prevState) => {
      return {
        ...prevState,
        isModalOpen: false,
      };
    });
  };

  const handleSubmit = (notice: NoticeDataType): void => {
    saveNotice(notice, handleAlert);
    setupdateNoticeCheck((val) => {
      return !val;
    });
    handleClose();
  };

  const handleChange = (event: ChangeEvent<HTMLTextAreaElement>): void => {
    const { name, value }: { name: string; value: string } = event.target;
    setSelectedNotice({
      ...selectedNotice,
      notice: {
        ...selectedNotice.notice,
        [name]: value,
      },
    });
  };

  return (
    <>
      <Paper className="w-full pb-8 md:w-[48%] lg:w-1/2 px-0 h-1/2 md:h-full flex-grow overflow-hidden rounded-xl">
        <Box className="mx-8 my-4 mb-2.5 flex justify-between">
          <Typography className="text-2xl font-medium">Notice</Typography>
          <Button
            className="hover:bg-primary-dark "
            variant="contained"
            endIcon={<AddIcon />}
            onClick={(event) =>
              handleOpenNotice(event, "add", intitalNoticeState)
            }
          >
            Add
          </Button>
        </Box>
        {notices ? (
          <List className="w-full h-[90%] p-0 overflow-y-scroll">
            {notices.reverse().map((notice: NoticeDataType, index: number) => {
              return (
                <Fragment key={notice.id}>
                  <NoticeListItem
                    noticeNumber={index}
                    noticeData={notice}
                    handleOpenNotice={handleOpenNotice}
                    handleOpenConfirmationDialog={() => handleDialogClick(true)}
                    setDeleteNoticeId={setDeleteNoticeId}
                  />
                  {index < notices.length - 1 && <Divider className="mx-8" />}
                </Fragment>
              );
            })}
          </List>
        ) : (
          <>
            {[1, 2, 3].map((index: number) => {
              return (
                <Box key={index.toString()} className="mx-8 h-1/3 pt-4">
                  <Skeleton width="30%" />
                  <Skeleton width="10%" />
                  <Skeleton width="100%" height="80%" />
                </Box>
              );
            })}
          </>
        )}
      </Paper>
      <NoticeModal
        selectedNotice={selectedNotice!}
        handleClose={handleClose}
        handleSubmit={handleSubmit}
        handleChange={handleChange}
      />
      <ConfirmationModal
        isOpen={open}
        handleClose={() => handleDialogClick(false)}
        handleSubmit={handleDialogSubmit}
        title="Are You Sure ?"
        buttontext="delete"
        buttonType="delete"
      >
        <DialogContent className="padding-0">
          This notice will be deleted
        </DialogContent>
      </ConfirmationModal>
      {isOpen && (
        <AlertComponent
          severity={severity}
          message={message}
          handleClose={() => handleAlert(false)}
        />
      )}
    </>
  );
};

export default NoticeList;
