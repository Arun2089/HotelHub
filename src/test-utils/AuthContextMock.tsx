import { AuthContext, AuthContextType } from "@src/context/AuthContext";

jest.mock("@src/utils/index.ts", () => ({
  fetchData: jest.fn().mockResolvedValue({
    id: 1,
    email: "johnDoe@gmail.com",
    role: { type: "admin" },
    username: "John Doe",
  }),
}));

type PropsWithChildren = {
  children: React.ReactNode;
  value: Partial<AuthContextType>;
};

const defaultValues = {
  user: {
    id: 1,
    email: "johnDoe@gmail.com",
    role: "admin",
    name: "John Doe",
  },
  updateUserDetails: jest.fn(),
  getUser: jest.fn().mockResolvedValue(true),
  isLoading: false,
  handleLogout: jest.fn(),
  handleLogin: jest.fn(),
  handleGoogleLogin: jest.fn(),
};

const AuthProviderMock = ({ children, value }: PropsWithChildren) => {
  return (
    <AuthContext.Provider value={{ ...defaultValues, ...value }}>
      {children}
    </AuthContext.Provider>
  );
};
export default AuthProviderMock;
