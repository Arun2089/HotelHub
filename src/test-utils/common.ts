export const setTimeOutMock = (ms: number) =>
  new Promise((resolve) => setTimeout(resolve, ms));
