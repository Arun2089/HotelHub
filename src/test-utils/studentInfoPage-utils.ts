import { fireEvent } from "@testing-library/react";

export const mockStudent = () => {
  return {
    studentName: "",
    roomNumber: null,
    mobileNumber: "",
    guardianName: "",
    guardianPhoneNumber: "",
    address: "",
    email: "",
    password: "",
    confirmPassword: "",
    govId: "",
    govIdImage: "",
  };
};

export const mockHandleChange = jest.fn();
export const mockValidateError = jest.fn();
export const mockGetHelperText = jest.fn();
export const mockHandleClick = jest.fn();

export function changeMultipleFields(
  textFields: HTMLElement[],
  details: { [key: string]: string | number | null | undefined }
) {
  textFields.forEach((textField) => {
    const name = textField.getAttribute("name");
    if (name) {
      fireEvent.change(textField, {
        target: { value: details[name] },
      });
    }
  });
}

export const studentDetails: {
  [key: string]: string | number | null | undefined;
} = {
  studentName: "John Doe",
  email: "john@gmail.com",
  mobileNumber: "9994567890",
};

export const guardianDetials: {
  [key: string]: string | number | null | undefined;
} = {
  guardianName: "John Doe",
  guardianPhoneNumber: "9994567890",
  address: "123, Main Street, New York",
};

export const passwordDetails: {
  [key: string]: string | number | null | undefined;
} = {
  password: "12345678",
  confirmPassword: "12345678",
};

export const govermentidDetails: {
  [key: string]: string | number | null | undefined;
} = {
  govId: "Pan Card",
  govIdImage: "Base64 string",
};
