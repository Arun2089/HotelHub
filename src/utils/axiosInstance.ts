import axios, { AxiosInstance } from "axios";
import { BASE_URL } from "@constant/index";
import { deleteLocalStorage, getLocalStorage } from "@utils/index";

const axiosInstance = (): AxiosInstance => {
  const instance = axios.create({
    baseURL: BASE_URL,
  });
  instance.interceptors.request.use(
    (config) => {
      const token = getLocalStorage("jwtToken");
      if (token) config.headers.Authorization = `Bearer ${token}`;
      else window.location.href = "/login";
      return config;
    },
    (error) => Promise.reject(error)
  );

  instance.interceptors.response.use(
    (response) => response,
    (error) => {
      if (error.response.status === 401) {
        deleteLocalStorage("jwtToken");
        window.location.href = "/login";
      }
      return Promise.reject(error);
    }
  );

  return instance;
};

export default axiosInstance;
