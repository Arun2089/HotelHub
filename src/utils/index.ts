import { StudentInfoType } from "@ts/types";
import { STUDENT_INFO_URL } from "@constant/index";

export const fetchData = async (url: string) => {
  const response = await fetch(url);
  if (response.ok) {
    const data = await response.json();
    return data;
  } else {
    throw new Error(`Error in retrieving data from ${url}`);
  }
};
export const sendData = async (
  url: string,
  method: "POST" | "PUT",
  data: any //eslint-disable-line
): Promise<boolean> => {
  const response = await fetch(url, {
    method: method,
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify(data),
  });
  if (response.ok) {
    return true;
  } else {
    throw new Error(`Error in sending data`);
  }
};

export const deleteData = async (url: string): Promise<boolean> => {
  const response = await fetch(url, {
    method: "DELETE",
  });
  if (response.ok) {
    return true;
  } else {
    throw new Error(`Error in deleting data from ${url}`);
  }
};

export const checkEmailExists = async (
  email: string,
  id?: number
): Promise<boolean> => {
  try {
    const data = await fetchData(STUDENT_INFO_URL);
    if (id) {
      const emailExists = data.some(
        (student: StudentInfoType) =>
          student.email === email && student.id !== id
      );
      return emailExists;
    }
    const emailExists = data.some(
      (student: StudentInfoType) => student.email === email
    );
    return emailExists;
  } catch (error) {
    throw error as string;
  }
};

export const catchErrorMessage = (error: unknown) => {
  return error instanceof Error ? error.message : "An error occurred";
};
