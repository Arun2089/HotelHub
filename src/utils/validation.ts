function validateEmail(email: string) {
  const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
  if (email.trim() === "" || !emailRegex.test(email)) {
    return false;
  }
  return true;
}

function validatePhone(phone: string) {
  const phoneRegex = /^[6-9]\d{9}$/;
  if (phone.trim() === "" || !phoneRegex.test(phone)) {
    return false;
  }
  return true;
}

function validatePassword(password: string) {
  if (password.length < 8 || password.trim() === "") {
    return false;
  }
  return true;
}

function validateText(text: string) {
  const nameRegex = /^[A-Za-z\s]+$/;
  if (text.trim() === "" || !nameRegex.test(text)) {
    return false;
  }
  return true;
}

const validateError = (value: string, type: string) => {
  if (value === "") return false;
  switch (type) {
    case "text":
      return !validateText(value);
    case "email":
      return !validateEmail(value);
    case "tel":
      return !validatePhone(value);
    case "password":
    case "confirmPassword":
      return !validatePassword(value);
    default:
      return false;
  }
};
const getHelperText = (
  value: string,
  type: string,
  confirmPassword?: string
): string => {
  if (validateError(value, type)) {
    switch (type) {
      case "text":
        return "Name cannot have special characters";
      case "email":
        return "Enter a valid email";
      case "tel":
        return "Enter a valid phone number";
      case "password":
      case "confirmPassword": {
        return "Password must contain atleast 8 characters";
      }
      default:
        return "Invalid input";
    }
  } else if (type === "confirmPassword" && value && value !== confirmPassword) {
    return "Password does not match";
  } else return "";
};

export {
  validateEmail,
  validatePhone,
  validatePassword,
  validateText,
  validateError,
  getHelperText,
};
