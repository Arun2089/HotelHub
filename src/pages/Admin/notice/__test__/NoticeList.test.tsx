import {
  render,
  screen,
  fireEvent,
  waitFor,
  act,
} from "@testing-library/react";
import "@testing-library/jest-dom";
import AuthProviderMock from "@src/test-utils/AuthContextMock";
import NoticeList from "../NoticeList";
import { sendData, deleteData } from "@src/utils";
import { setTimeOutMock } from "@src/test-utils/common";

const mockHandleAlert = jest.fn();
const mockSetUpdateNoticeCheck = jest.fn();

const mockNotices = [
  {
    id: 1,
    title: "Notice 1",
    date: "2022-01-01",
    content: "This is notice 1",
  },
  {
    id: 2,
    title: "Notice 2",
    date: "2022-01-02",
    content: "This is notice 2",
  },
];

jest.mock("@src/utils/index.ts", () => ({
  ...jest.requireActual("@src/utils/index.ts"),
  sendData: jest.fn(),
  deleteData: jest.fn(),
  // todayDate: () => "2022-01-01",
}));

jest.mock("@src/hooks/useAlert.ts", () => {
  return jest.fn().mockImplementation(() => {
    return {
      alert: {
        isOpen: false,
        message: "",
        severity: "",
      },
      handleAlert: mockHandleAlert,
    };
  });
});

const clickButton = (buttonTestId: string, buttonName: string) => {
  const button = screen.getAllByTestId(buttonTestId);
  fireEvent.click(button[1]);
  const openedNoticeModal = screen.getByRole("dialog");
  expect(openedNoticeModal).toBeInTheDocument();
  const confirmButton = screen.getByRole("button", { name: buttonName });
  act(() => {
    fireEvent.click(confirmButton);
  });
};

const checkModalBeforeClick = () => {
  const noticeModal = screen.queryByRole("dialog");
  expect(noticeModal).not.toBeInTheDocument();
};

describe("NoticeList", () => {
  beforeEach(() => {
    render(
      <AuthProviderMock value={{ isLoading: false }}>
        <NoticeList notices={mockNotices} />
      </AuthProviderMock>
    );
  });
  afterEach(() => {
    jest.clearAllMocks();
  });

  it("snapshot test", () => {
    const { asFragment } = render(
      <AuthProviderMock value={{ isLoading: false }}>
        <NoticeList notices={mockNotices} />
      </AuthProviderMock>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the component with notices", () => {
    const notice1 = screen.getByText("Notice 1");
    const notice2 = screen.getByText("Notice 2");
    expect(notice1).toBeInTheDocument();
    expect(notice2).toBeInTheDocument();
  });

  it("renders the component without notices", () => {
    render(
      <AuthProviderMock value={{ isLoading: false }}>
        <NoticeList notices={[]} />
      </AuthProviderMock>
    );
    const noNoticesMessage = screen.getByText("No notices");
    expect(noNoticesMessage).toBeInTheDocument();
  });

  it("opens the notice modal when 'Add' button is clicked", () => {
    const noticeModal = screen.queryByRole("dialog");
    expect(noticeModal).not.toBeInTheDocument();
    const addButton = screen.getByText("Add");
    fireEvent.click(addButton);
    const openedNoticeModal = screen.getByRole("dialog");
    expect(openedNoticeModal).toBeInTheDocument();
  });

  it("saveNotice works when Save button is Clicked and response is ok", async () => {
    (sendData as jest.Mock).mockResolvedValue({
      status: 201,
      message: "Notice added",
    });
    checkModalBeforeClick();
    clickButton("EditOutlinedIcon", "Save");
    await act(async () => {
      await setTimeOutMock(1000);
    });
    await waitFor(() => {
      expect(sendData).toHaveBeenCalledTimes(1);
      expect(mockHandleAlert).toHaveBeenCalledWith(
        true,
        "Notice updated",
        "success"
      );
    });
  });

  it("saveNotice works when Save button is Clicked and response is not ok", async () => {
    (sendData as jest.Mock).mockRejectedValue({
      status: 400,
      message: "Notice not added",
    });
    checkModalBeforeClick();
    clickButton("EditOutlinedIcon", "Save");
    await act(async () => {
      await setTimeOutMock(1000);
    });
    await waitFor(() => {
      expect(sendData).toHaveBeenCalledTimes(1);
      expect(mockHandleAlert).toHaveBeenCalledWith(
        true,
        "An error occurred",
        "error"
      );
    });
  });

  it("deleteNotice works when Delete button is Clicked and response is ok", async () => {
    (deleteData as jest.Mock).mockResolvedValue({
      status: 202,
      message: "Notice deleted",
    });
    checkModalBeforeClick();
    clickButton("DeleteOutlineOutlinedIcon", "delete");
    await act(async () => {
      await setTimeOutMock(1000);
    });
    await waitFor(() => {
      expect(deleteData).toHaveBeenCalledTimes(1);
      expect(mockSetUpdateNoticeCheck).toHaveBeenCalledTimes(1);
      expect(mockHandleAlert).toHaveBeenCalledWith(
        true,
        "Notice deleted",
        "success"
      );
    });
  });

  it("deleteNotice works when Delete button is Clicked and response is not ok", async () => {
    (deleteData as jest.Mock).mockRejectedValue({
      status: 400,
      message: "Notice not deleted",
    });
    checkModalBeforeClick();
    clickButton("DeleteOutlineOutlinedIcon", "delete");

    await act(async () => {
      await setTimeOutMock(1000);
    });

    await waitFor(() => {
      expect(deleteData).toHaveBeenCalledTimes(1);
      expect(mockSetUpdateNoticeCheck).toHaveBeenCalledTimes(1);
      expect(mockHandleAlert).toHaveBeenCalledWith(
        true,
        "An error occurred",
        "error"
      );
    });
  });
});

describe("NoticeList for Students", () => {
  it("does not render Add Button when user is Student", async () => {
    render(
      <AuthProviderMock
        value={{
          user: {
            id: 1,
            email: "johnDoe@gmail.com",
            role: "",
            name: "John Doe",
          },
        }}
      >
        <NoticeList notices={mockNotices} />
      </AuthProviderMock>
    );
    await waitFor(() => {
      const addButton = screen.queryByText("Add");
      expect(addButton).not.toBeInTheDocument();
    });
  });
});
