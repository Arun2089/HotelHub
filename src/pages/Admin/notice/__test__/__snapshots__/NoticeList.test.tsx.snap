// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoticeList snapshot test 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 w-full pb-8 md:w-[48%] lg:w-1/2 px-0 h-[50vh] md:h-full flex-grow overflow-hidden rounded-xl css-1ps6pg7-MuiPaper-root"
    data-testid="notice-component"
  >
    <div
      class="mx-8 my-4 mb-2.5 flex justify-between MuiBox-root css-0"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 text-xl md:text-2xl font-medium css-ahj2mt-MuiTypography-root"
      >
        Notice
      </p>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium hover:bg-primary-dark  css-sghohy-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Add
        <span
          class="MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="AddIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <ul
      class="MuiList-root MuiList-padding w-full h-[90%] p-0 overflow-y-scroll css-h4y409-MuiList-root"
    >
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-alignItemsFlexStart px-8 cursor-pointer py-1hover:bg-gray-100 css-hhvo2v-MuiListItem-root"
      >
        <div
          class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            <div
              class="flex justify-between MuiBox-root css-0"
            >
              <div
                class="MuiBox-root css-0"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 font-semibold text-sm md:text-base css-ahj2mt-MuiTypography-root"
                >
                  <span
                    class="pr-1"
                  >
                    1.
                  </span>
                  Notice 1
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 text-gray-500 text-[0.7rem] md:text-xs ml-[15px] css-ahj2mt-MuiTypography-root"
                >
                  2022-01-01
                </p>
              </div>
              <div
                class="MuiBox-root css-0"
                id="notice-icon-btns"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeSmall text-sm css-2rshpx-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                    data-testid="EditOutlinedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="m14.06 9.02.92.92L5.92 19H5v-.92l9.06-9.06M17.66 3c-.25 0-.51.1-.7.29l-1.83 1.83 3.75 3.75 1.83-1.83c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.2-.2-.45-.29-.71-.29zm-3.6 3.19L3 17.25V21h3.75L17.81 9.94l-3.75-3.75z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorError MuiIconButton-sizeSmall css-1mk6yu1-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                    data-testid="DeleteOutlineOutlinedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM8 9h8v10H8V9zm7.5-5-1-1h-5l-1 1H5v2h14V4h-3.5z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </span>
          <p
            class="MuiTypography-root MuiTypography-body1 line-clamp-3 text-sm md:text-base ml-[15px] mr-2.5 mt-1 css-ahj2mt-MuiTypography-root"
          >
            This is notice 1
          </p>
        </div>
      </li>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth mx-8 css-9mgopn-MuiDivider-root"
      />
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-alignItemsFlexStart px-8 cursor-pointer py-1hover:bg-gray-100 css-hhvo2v-MuiListItem-root"
      >
        <div
          class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            <div
              class="flex justify-between MuiBox-root css-0"
            >
              <div
                class="MuiBox-root css-0"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 font-semibold text-sm md:text-base css-ahj2mt-MuiTypography-root"
                >
                  <span
                    class="pr-1"
                  >
                    2.
                  </span>
                  Notice 2
                </p>
                <p
                  class="MuiTypography-root MuiTypography-body1 text-gray-500 text-[0.7rem] md:text-xs ml-[15px] css-ahj2mt-MuiTypography-root"
                >
                  2022-01-02
                </p>
              </div>
              <div
                class="MuiBox-root css-0"
                id="notice-icon-btns"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeSmall text-sm css-2rshpx-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                    data-testid="EditOutlinedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="m14.06 9.02.92.92L5.92 19H5v-.92l9.06-9.06M17.66 3c-.25 0-.51.1-.7.29l-1.83 1.83 3.75 3.75 1.83-1.83c.39-.39.39-1.02 0-1.41l-2.34-2.34c-.2-.2-.45-.29-.71-.29zm-3.6 3.19L3 17.25V21h3.75L17.81 9.94l-3.75-3.75z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorError MuiIconButton-sizeSmall css-1mk6yu1-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
                    data-testid="DeleteOutlineOutlinedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM8 9h8v10H8V9zm7.5-5-1-1h-5l-1 1H5v2h14V4h-3.5z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
            </div>
          </span>
          <p
            class="MuiTypography-root MuiTypography-body1 line-clamp-3 text-sm md:text-base ml-[15px] mr-2.5 mt-1 css-ahj2mt-MuiTypography-root"
          >
            This is notice 2
          </p>
        </div>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
